CMake/PackageConfig.cmake

```diff
@@ -1,5 +1,5 @@
 add_library(HighFive INTERFACE)
-target_link_libraries(HighFive INTERFACE ${HDF5_LIBRARIES})
+target_link_libraries(HighFive INTERFACE ${HDF5_LIBRARIES} Boost::serialization)
```

Comment:
Boost is optional. Im afraid if the target doesnt exist we might have problems
Why do we need boost::serialization?

Reply:
boost::serialization for https://github.com/BlueBrain/HighFive/blob/74250cf2f2a82a2b9af2f00f93475c431a2d7e09/include/highfive/H5DataSpace.hpp#L21
But yes we need to do it optional.
I will look tomorrow to get PackageConfig back in CMakeLists, it will be more straight forward and easy.